tests/suites/test_suite_net.function

```diff
@@ -0,0 +1,121 @@
+/* BEGIN_HEADER */
+
+#include "mbedtls/net_sockets.h"
+
+#if defined(unix) || defined(__unix__) || defined(__unix) || \
+    defined(__APPLE__) || defined(__QNXNTO__) || \
+    defined(__HAIKU__) || defined(__midipix__)
+#define MBEDTLS_PLATFORM_IS_UNIXLIKE
+#endif
+
+#if defined(MBEDTLS_PLATFORM_IS_UNIXLIKE)
+#include <sys/fcntl.h>
+#include <sys/resource.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#endif
+
+/* END_HEADER */
+
+/* BEGIN_DEPENDENCIES
+ * depends_on:MBEDTLS_NET_C
+ * END_DEPENDENCIES
+ */
+
+/* BEGIN_CASE */
+void context_init_free( int reinit )
+{
+    mbedtls_net_context ctx;
+
+    mbedtls_net_init( &ctx );
+    mbedtls_net_free( &ctx );
+
+    if( reinit )
+        mbedtls_net_init( &ctx );
+    mbedtls_net_free( &ctx );
+
+    /* This test case always succeeds, functionally speaking. A plausible
+     * bug might trigger an invalid pointer dereference or a memory leak. */
+    goto exit;
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PLATFORM_IS_UNIXLIKE */
+void poll_beyond_fd_setsize( )
+{
+    /* Test that mbedtls_net_poll does not misbehave when given a file
+     * descriptor beyond FD_SETSIZE. This code is specific to platforms
+     * with a Unix-like select() function. */
+
+    struct rlimit rlim_nofile;
+    int restore_rlim_nofile = 0;
+    /* We want to have an open file descriptor >= FD_SETSIZE. This requires
+     * the file descriptor limit to be at least FD_SETSIZE + 1. Overshoot
+     * slightly in case the highest possible file descriptor(s) were already
+     * open, but not so much that we're likely to be blocked by a hard limit. */
+    rlim_t min_wanted_nofile = FD_SETSIZE + 4;
+    fd_set open_fds;
+    int ret;
+    int fd = -1;
+    mbedtls_net_context ctx;
+    uint8_t buf[1];
+
+    mbedtls_net_init( &ctx );
+    FD_ZERO( &open_fds );
```

Comment:
On some platforms, such as Gnu-Linux (x86-64), `FD_ZERO` appears to be implemented in assembly, hence not accessible to UBSan's compile-time instrumentation. As a result, the CI fails in the memsan components (but not the the valgrind tests, runtime instrumentation is superior in that respect). We have the same issue in `net_sockets.c` so I suggest fixing it the same way:
```diff
diff --git a/tests/suites/test_suite_net.function b/tests/suites/test_suite_net.function
index 7b8f4134b1f5..45d47c000514 100644
--- a/tests/suites/test_suite_net.function
+++ b/tests/suites/test_suite_net.function
@@ -62,6 +62,13 @@ void poll_beyond_fd_setsize( )
     uint8_t buf[1];
 
     mbedtls_net_init( &ctx );
+#if defined(__has_feature)
+#if __has_feature(memory_sanitizer)
+    /* Ensure that UBSan considers open_fds as initialized even on platforms
+     * such as Glibc/x86_64 where FD_ZERO is implemented in assembly. */
+    memset( &open_fds, 0, sizeof( open_fds ) );
+#endif
+#endif
     FD_ZERO( &open_fds );
 
     /* On many systems, by default, the maximum number of open file descriptors
```
I confirmed that this fixes the failure locally for me.

Reply:
I wonder why we don't need this in `mbedtls_net_recv_timeout` as well.

Reply:
Moot now that I've changed the test code to dup to the desired file descriptor, rather than open files until we reach the desired range. (Like throwing darts to the floor until the stack reaches as high as the target…)